server/src/test/java/org/opensearch/common/time/DateFormattersTests.java

```diff
@@ -461,6 +475,132 @@ public void testIso8601Parsing() {
         formatter.format(formatter.parse("2018-05-15T17:14:56,123456789+01:00"));
     }
 
+    public void testRFC339Parsing() {
+        DateFormatter formatter = DateFormatters.forPattern("rfc3339");
+
+        // timezone not allowed with just date
+        formatter.format(formatter.parse("2018"));
+        formatter.format(formatter.parse("2018-05"));
+        formatter.format(formatter.parse("2018-05-15"));
+
+        formatter.format(formatter.parse("2018-05-15T17:14Z"));
+        formatter.format(formatter.parse("2018-05-15T17:14z"));
+        formatter.format(formatter.parse("2018-05-15T17:14+01:00"));
+        formatter.format(formatter.parse("2018-05-15T17:14-01:00"));
+
+        formatter.format(formatter.parse("2018-05-15T17:14:56Z"));
+        formatter.format(formatter.parse("2018-05-15T17:14:56z"));
+        formatter.format(formatter.parse("2018-05-15T17:14:56+01:00"));
+        formatter.format(formatter.parse("2018-05-15T17:14:56-01:00"));
+
+        // milliseconds can be separated using comma or decimal point
+        formatter.format(formatter.parse("2018-05-15T17:14:56.123Z"));
```

Comment:
Why are we not adding any assertions here?

Reply:
> Why are we not adding any assertions here?

Test relies on successful parsing without exceptions and then successful conversion back to string

>A heads up that comma is not allowed in RFC-3339, opposed to ISO-8601. The context is a little unclear to me, so I wanted to just mention it.

Yes, the parser is lenient in several ways w.r.t. to rfc3339 but it is rfc3339 compatible. Making it lenient is intentional to increase usability